### ISSUE TITLE
Rich error messages

### ISSUE THREAD

**Issue by gsomix** [ForNeVeR/wpf-math#34]
We should to revise all parser error messages and make them more concise and rich. E.g. add info about index of symbol where error was occured.

**Comment by alexreg** [comment]
Agreed. A consistent writing style would be nice too. If you would like my input here, I'd be happy to provide it, although I've seen enough to know both of your English is very good, of course!
